### PR TITLE
DietPi-Software | Aria2: Installer rework

### PR DESCRIPTION
**Pull request by msongz**
to avoid losing tasks after reboot the device

**Status**: Testing
- [x] should adding /var/lib/dietpi/dietpi-software/installed/aria2.session first to make it works?

### CODE REVIEW

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -11139,9 +11144,7 @@ bt-max-peers=$(Optimise_BitTorrent 2)
 listen-port=6881-6999
 check-certificate=false
 
-max-overall-upload-limit=0
 max-upload-limit=0
-max-overall-download-limit=0
 max-download-limit=0
```
Comment (MichaIng) [inline dietpi/dietpi-software]: @msongz 
Why you removed the overall limits? They default to `0` anyway, but this is true for the per-torrent limits as well.
#### EDIT: Ah I see the doubled entries.

**Comment by MichaIng** [comment]
@msongz 
Many thanks for this, looks very reasonable.
What do you think about moving the aria2.conf and session files to `/mnt/dietpi_userdata`? E.g. in other downloaders we save the session file right in `/mnt/dietpi_userdata/downloads/`:
- qBittorrent save and tmp paths: https://github.com/MichaIng/DietPi/blob/dev/dietpi/dietpi-software#L10286-L10288
- ruTorrent session dir: https://github.com/MichaIng/DietPi/blob/dev/dietpi/dietpi-software#L10541
- and others as well.

E.g.
```
/mnt/dietpi_userdata/aria2/aria2.conf
/mnt/dietpi_userdata/downloads/aria2.session
```
Or both in `/aria2/`, not sure what is best to go with here, however I like the idea to keep configs and states in userdata, since this can be easily moved to external drive and is there safer against data corruption/broken SDcard and can be easily migrated to a new system.

> should be add a file : `/var/lib/dietpi/dietpi-software/installed/aria2.session` first to make it works?

Did you test if there are any errors when starting aria2c with `input-file` pointing to non-existent file? Probably a warning, but I assume it would simply start and then create the file on first periodic save interval.

File: dietpi/dietpi-software (hunk 1 above)
Comment (msongz) [reply dietpi/dietpi-software]: 😄

**Comment by msongz** [comment]
`Exception: [download_helper.cc:563] errorCode=1 Failed to open the file /var/lib/dietpi/dietpi-software/installed/aria2.session, cause: File not found or it is a directory`
after I test update the config with the session file and restart the service with systemctl, it prompts an error  and prevents the service start.
so the session file may be created first to make this works

**Comment by msongz** [comment]
regarding the config and the session file path, I love your point

**Comment by MichaIng** [comment]
@msongz 
Perfect, I work on quite some rework now, also to run aria2 as user "aria2", group "dietpi" and umask 002, so media player run users within dietpi group can directly access aria2 downloads.

**Comment by MichaIng** [comment]
@msongz 
Okay, changes done. What you think about it?

Requires some patch on update to create new user and move config file.

**Comment by msongz** [comment]
nice work! would like to test it ASAP. 😄

**Comment by MichaIng** [comment]
@msongz 
Okay will merge for testing, especially since your branch is based on master while we should test with current dev branch around 😉.

#### EDIT: Whoopsie, dumb syntax/forgotten username 😄: https://github.com/MichaIng/DietPi/commit/1a5d29d9713c133dafddebdaa8745e21153b7c2b
#### EDIT2: Log file needs to be pre-created: https://github.com/MichaIng/DietPi/commit/7573a511156021cafcce5d6334b6561477af8f53

Okay after above commits it works well here now, fresh installs and reinstalls.

Changelog: https://github.com/MichaIng/DietPi/commit/1b9597399ab5a671e808170dbd3603ee857c8ed1

On v6.26 update, apply new session store/resume feature but preserve custom user settings, if applied: https://github.com/MichaIng/DietPi/commit/58455252a7a81dd0de1894651dbd7201b78d1920

Another fix for web UI install:
- New dir was moved **into** the existing dir, thus nothing was updated.
- Now existing dir is fully removed, since those files are never written to, settings are stored in browser/client only: https://github.com/MichaIng/DietPi/commit/44ce25bc7b55197e159bcf7f62a048ad6b886d52

**Pull request merged**